server/tools/hatohol-db-initiator.in

```diff
@@ -235,14 +202,17 @@ def get_default_sql_dir():
     return os.path.dirname(__file__) + '/../share/hatohol/sql'
 
 
-def get_target_list():
-    target_list = []
-    [target_list.append(elem['target']) for elem in sql_file_list]
-    return target_list
 
 if __name__ == '__main__':
     init_config_map()
 
+    default_sql_dir = get_default_sql_dir()
+    if "--sql-dir" in sys.argv:
+        target_dir = sys.argv[sys.argv.index("--sql-dir")+1]
+    else:
+        target_dir = default_sql_dir
+    target_list = os.listdir(target_dir)
```

Comment:
This line should be in the 'if args.show_target' block.
os.listdir() is always invoked even when it's unnecessary.